src/components/dev-hub/tab-context.js

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect, useReducer } from 'react';
+import { getLocalValue, setLocalValue } from '../../utils/browser-storage';
+
+const defaultContextValue = {
+    activeTabs: {},
+    setActiveTab: () => {},
+};
+
+const TabContext = React.createContext(defaultContextValue);
+
+const reducer = (prevState, { name, value }) => ({
```

Comment:
Why do we keep track of prevState here?

Reply:
Ah that way if we want to add new tabs in we don't overwrite the old one in this object

Reply:
This piece of state actually wraps the entire app so preferences are kept between articles, and different articles could have different sets of tabs

Reply:
Okay, I see! I totally misunderstood what this was doing. Maybe it would be helpful to rename the name/value variables to better describe what we are actually storing here?

Reply:
Yep! Great idea, I went with `tabset` to hopefully make that more clear as well as `tabSelectionPreferences`